source/MRMesh/MRDeferred.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <utility>
+
+// TODO: move to separate header
+#define MR_CONCAT_IMPL( A, B ) A##B
+#define MR_CONCAT( A, B ) MR_CONCAT_IMPL( A, B )
+
+namespace MR
+{
+
+template <typename F>
+class DeferredFunction
+{
+    F f_;
+
+public:
+    DeferredFunction( F&& f )
+        : f_( std::forward<F>( f ) )
+    {
+        //
+    }
+    ~DeferredFunction()
+    {
+        f_();
+    }
+
+    DeferredFunction( const DeferredFunction& ) = delete;
+    DeferredFunction( DeferredFunction&& ) = delete;
+    DeferredFunction& operator =( const DeferredFunction& ) = delete;
+    DeferredFunction& operator =( DeferredFunction&& ) = delete;
+};
+
+template <typename F>
+DeferredFunction<F> defer( F&& f )
```

Comment:
Can we use https://en.cppreference.com/w/cpp/language/class_template_argument_deduction instead additional function?

Reply:
Fixed.

Reply:
I mean with CTAD addition function `defer` is not necessary